src/isDeepEqual.ts

```diff
@@ -174,3 +178,43 @@ function isDeepEqualMaps(
 
   return true;
 }
+
+function isDeepEqualSets(
+  data: ReadonlySet<unknown>,
+  other: ReadonlySet<unknown>,
+): boolean {
+  if (data.size !== other.size) {
+    return false;
+  }
+
+  // TODO: Once we bump our typescript target we can iterate over the map keys and values directly.
+  const dataArr = Array.from(data.values());
+  const otherMap = arrayToMap(Array.from(other.values()));
+  const otherKeys = Array.from(otherMap.keys());
+
+  let isFound = false;
+
+  for (const dataItem of dataArr) {
+    for (const otherKey of otherKeys) {
+      if (isDeepEqualImplementation(dataItem, otherMap.get(otherKey))) {
+        isFound = true;
+        otherMap.delete(otherKey);
+        break;
+      }
+    }
+    if (!isFound) {
+      return false;
+    }
+    isFound = false;
```

Comment:
scope isFound to the outer loop, it's only used within it's scope, and that way you don't need to reset it to `false`.
```suggestion
  for (const dataItem of dataArr) {
    let isFound = false;

    for (const otherKey of otherKeys) {
      if (isDeepEqualImplementation(dataItem, otherMap.get(otherKey))) {
        isFound = true;
        otherMap.delete(otherKey);
        break;
      }
    }
    if (!isFound) {
      return false;
    }
```